File: android_host_app/app/src/main/java/com/example/android_host_app/KycLogStore.kt
package com.example.android_host_app

object KycLogStore {

    data class LogEntry(
        val type: String,
        val step: String?,
        val message: String,
        val meta: String?,
        val timestampMillis: Long
    )

    private val _events = mutableListOf<LogEntry>()

    val events: List<LogEntry>
        get() = _events.toList()

    fun add(entry: LogEntry) {
        // latest first
        _events.add(0, entry)
    }

    fun clear() {
        _events.clear()
    }
}

File: ios_host_app/ios_host_app/KycLogStore.swift
import Foundation

struct KycLogEntry {
    let type: String
    let step: String?
    let message: String
    let meta: String?
    let timestamp: Date
}

final class KycLogStore {

    static let shared = KycLogStore()
    private init() {}

    private var items: [KycLogEntry] = []

    var events: [KycLogEntry] {
        return items
    }

    func add(_ entry: KycLogEntry) {
        // newest first – like Android
        items.insert(entry, at: 0)
    }

    func clear() {
        items.removeAll()
    }
}
